var/spack/repos/builtin/packages/hip/package.py

```diff
@@ -606,3 +607,68 @@ def cmake_args(self):
             args.append("-DCMAKE_INSTALL_LIBDIR=lib")
 
         return args
+
+    test_src_dir = "samples"
+
+    @run_after("install")
+    def cache_test_sources(self):
+        """Copy the tests source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        if self.spec.satisfies("@:5.1.0"):
+            return
+        self.cache_extra_test_sources([self.test_src_dir])
+
+    def test_samples(self):
+        # configure, build and run all hip samples
+        if self.spec.satisfies("@:5.1.0"):
+            print("Skipping: stand-alone tests")
+            return
```

Comment:
Looks like I need to update the documentation.  In the meantime, you could change this to raise the new `SkipTest` exception here, along the lines:

```suggestion
        if self.spec.satisfies("@:5.1.0"):
            raise SkipTest("Test is only available for specs after version 5.1.0")
```

This is the new way of flagging a test case as being skipped for CDash (so CI) reporting.